src/core/manager-serialize.c

```diff
@@ -478,16 +470,13 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                         }
 
                 } else if ((val = startswith(l, "user-lookup="))) {
-                        int fd0, fd1;
-
-                        if (sscanf(val, "%i %i", &fd0, &fd1) != 2 || fd0 < 0 || fd1 < 0 || fd0 == fd1 || !fdset_contains(fds, fd0) || !fdset_contains(fds, fd1))
-                                log_notice("Failed to parse user lookup fd, ignoring: %s", val);
-                        else {
-                                m->user_lookup_event_source = sd_event_source_disable_unref(m->user_lookup_event_source);
-                                safe_close_pair(m->user_lookup_fds);
-                                m->user_lookup_fds[0] = fdset_remove(fds, fd0);
-                                m->user_lookup_fds[1] = fdset_remove(fds, fd1);
-                        }
+
+                        m->user_lookup_event_source = sd_event_source_disable_unref(m->user_lookup_event_source);
```

Comment:
other sources above are deref'ed only if deserializing succeeds, this does it unconditionally, is that intended?

Reply:
we sometimes do it like this and sometimes the other way, if you look through the codebase...

i think as long as we don't leak memory either is fine, it shouldn't happen anyway that we run into an issue here...